Automattic/jetpack
# Sync: Add links when expanding posts at a later stage

Fixes https://github.com/Automattic/vulcan/issues/297

## Proposed changes:

When syncing posts, we used to expand them via the `filter_post_content_and_add_links`. This was originally done before sending, but after dedicated sync was created, we decided to move it to before enqueuing the actions.

Since sometimes the links like [permalink](https://github.com/Automattic/jpop-issues/issues/8721) are not available.  The approach here is to separate the filtering part from the adding links.

The expectation is that adding the links before we send the action, as we used to do will allow us to fetch the proper links.

The possible problem here is that with Dedicated Sync enabled, we would be fetching those fields on init because this is [when we spawn the next dedicated sync request](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/sync/src/class-actions.php#L114). 

### Other information:


## Does this pull request change what data or activity we track or use?


No
## Testing instructions:





The initial idea here is to test that we are still properly populating the links.

In a connected Jetpack site with this patch.
- Create a post and publish it.
- Check that the sync action `jetpack_sync_save_post` The second one, when the post is published, has the proper permalink.
- Update the post.
- Check that the permalink is still showing in the action.

Now test with a plugin that will change the permalink like Yoast SEO.
- Create a post and publish it.
- Check that the sync action `jetpack_sync_save_post` 
- Update the post, but this time by updating Yoast SEO slug parameter which will change the permalink.
- Check that the permalink in the `jetpack_sync_save_post`  has changed.

Bonus points if tested with WPML plugin as described in the links inside the [issue](https://github.com/Automattic/vulcan/issues/297)
